### DISCUSSION TITLE
不使用det 手动给boxes rec的时候只能识别出一个框内的文字

### DISCUSSION THREAD

**Discussion by poison857** [PaddlePaddle/PaddleOCR#14082] [Q&A]
手动给的roi，框的位置 和 crop_image都看过 是正常的  请问一下大家rec的时候为啥就只能识别出一个框内的文字 其余框内的就不行

**Comment by GreatV** [comment]
您好，根据您的描述，您手动提供了ROI（感兴趣区域），并检查了框的位置和裁剪后的图像，一切看起来正常。但在进行文字识别时，只能识别出一个框内的文字，其他框内的文字无法识别。

可能的原因和解决方法如下：

1. **循环处理问题**：请检查您的代码，确认是否对所有的ROI进行了循环处理。可能您的代码只对第一个ROI进行了识别，没有遍历处理所有的框。

2. **ROI列表格式**：确保您传递给识别函数的ROI是一个包含所有框的列表或数组，而不是单个框。识别函数应能接收并处理多个ROI。

3. **图像预处理**：虽然您检查了裁剪后的图像，但可能某些图像的质量不佳，影响了识别效果。建议对裁剪后的图像进行预处理，例如灰度化、二值化、去噪等，提高图像质量。

4. **识别函数限制**：有些OCR识别函数默认只处理一张图像。请确认您的识别函数是否支持批量处理，或者需要在循环中逐一处理每个裁剪后的图像。

5. **错误日志查看**：查看程序运行时的日志或控制台输出，看看是否有错误提示。这可能会给您提供更多有关问题的信息。

6. **模型或配置问题**：如果您使用的是特定的OCR模型，确保模型配置正确，且适用于您要识别的文字类型。某些模型可能对特定语言或字符集有优化。

7. **位置坐标准确性**：再次确认所有ROI的坐标是否准确无误，避免坐标错误导致裁剪区域为空白或错误。

8. **并发处理问题**：如果您使用多线程或并发处理，可能存在资源竞争或覆盖的问题。尝试在单线程环境下运行，看看问题是否仍然存在。

建议您逐步调试程序，可以在处理每个ROI时输出一些调试信息，确认每个步骤都按预期执行。希望这些建议能帮助您解决问题！

**Reply by poison857** [reply]
你好 我的应用场景是 我做了一个dll，并重写了一个[ocr函数](std::vector<OCRPredictResult> PPOCR::ocr(cv::Mat img, std::vector<std::vector<std::vector<int>>> boxes, bool det, bool rec, bool cls))。其中det和cls修改为了false，我会在qt里调用这个dll的接口返回char*
现在遇到的问题是 同一张图像，我只框一个roi区域 原图中每个字符段都能识别，但是我框两个及以上，就只能识别出一个框里的，并且输出的结果和我给进去的图像顺序是被交换了的
![微信截图_20241025092143](https://github.com/user-attachments/assets/4e01c596-8ff8-4de4-84ca-75ae569297b5)
![微信截图_20241025092325](https://github.com/user-attachments/assets/19cb9287-7999-4293-8691-8444f987060e)
![微信截图_20241025092344](https://github.com/user-attachments/assets/8afe553e-f244-427f-9074-e7d84981e022)
![微信截图_20241025092714](https://github.com/user-attachments/assets/bc5e6ee8-63d8-435f-99ad-9fa4cf8742be)

**Reply by GreatV** [reply]
你用python版本的试试看

**Reply by poison857** [reply]
好的 我试试。请问有没有可能是args.cpp里的某个参数需要修改啊 我现在全是默认的参数
